### ISSUE TITLE
Instant loading: internal anchor links don't work in Safari

### ISSUE THREAD

**Issue by thesuperzapper** [squidfunk/mkdocs-material#6334]
### Bug description

In MkDocs Material `9.4.8`, it seems that internal heading anchor links don't work properly on Safari when using `navigation.instant`.

Note, this reliably happens, but sometimes you might need to clear your cache first.

### Related links

- https://github.com/squidfunk/mkdocs-material/issues/3797

### Reproduction

[9.4.8-safari-anchor-links.zip](https://github.com/squidfunk/mkdocs-material/files/13355433/9.4.8-safari-anchor-links.zip)

### Steps to reproduce

1. Use my reproduction.
2. Open the website in safari.
3. Click the link on the homepage.
4. Notice that you are at the top of the page, not scrolled to "Section 2" 

(Also, I am not sure if it's related, but the table of contents sidebar will be partially cut off until you scroll the page)

<img width="321" alt="Screenshot 2023-11-14 at 11 54 13" src="https://github.com/squidfunk/mkdocs-material/assets/5735406/293fed61-7b82-4c0d-9ee4-d5f74285a633">

### Browser

Safari

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting. Could I ask you to try to bisect the version that introduced the problem? As you noted, we fixed it in #3797, so you would do us a great favor in helping isolate the bad commit ☺️

**Comment by thesuperzapper** [comment]
Well, I was also seeing this on `9.2.3`, so I am not sure if that helps.

**Comment by thesuperzapper** [comment]
Another interesting issue which might be related (occurs on all browsers):

1. Click a internal link
2. Click the browser's back button
3. Click the same link again (fails to do anything)

**Comment by PaulSonOfLars** [comment]
Can confirm that when using `navigation.instant` and `navigation.footer`, the footer links will stop working after pressing the browser's "back" button

**Comment by thesuperzapper** [comment]
@PaulSonOfLars @squidfunk I have raised a separate issue for the more serious problem I raised in https://github.com/squidfunk/mkdocs-material/issues/6334#issuecomment-1811778534.


New issue: https://github.com/squidfunk/mkdocs-material/issues/6345

**Comment by squidfunk** [comment]
Fixed in https://github.com/squidfunk/mkdocs-material-insiders/pull/79 – the fixes are currently only in Insiders, so anybody having access can already test them. I still need to do some minor refactoring that shouldn't impact the functionality, and after that, I'll backport the fixes to the community edition. I just wanted to give them into the hands of users as soon as possible. This should resolve all known problems with instant loading, at least I've tested this and all other issues.
